packages/api-client-core/src/urlParamExchange.ts

```diff
@@ -0,0 +1,34 @@
+import type { Exchange } from "@urql/core";
+import type { DocumentNode, OperationDefinitionNode } from "graphql";
+import { onPush, pipe } from "wonka";
+
+export const graphqlDocumentName = (doc: DocumentNode) => {
+  const lastDefinition: OperationDefinitionNode | undefined = [...doc.definitions]
+    .reverse()
+    .find((definition) => definition.kind == "OperationDefinition") as any;
+  if (lastDefinition) {
+    if (lastDefinition.name) {
+      return lastDefinition.name.value;
+    }
+    const firstSelection = lastDefinition.selectionSet.selections.find((node) => node.kind == "Field") as any;
+    return firstSelection.name.value;
+  }
+};
+
+export const urlParamExchange: Exchange = ({ forward }) => {
+  return (ops$) =>
+    pipe(
+      ops$,
+      onPush((op) => {
```

Comment:
was wondering what the difference between `tap` and `onPush` was and turns out they're the same.